### ISSUE TITLE
Uncaught ReferenceError when running bundle created with Deno 1.5.x

### ISSUE THREAD

**Issue by Nitive** [denoland/deno#8399]
Hello! I'm experiencing the same issue as #8211, but with encoding/yaml module from standard library

**index.ts**
```js
import * as yaml from "https://deno.land/std@0.78.0/encoding/yaml.ts"
console.log(yaml && 1)
```

**Dockerfile**
```Dockerfile
FROM hayd/deno:alpine-1.5.2 AS build

WORKDIR /app

COPY index.ts ./

# no error
RUN deno run index.ts

RUN deno bundle ./index.ts index.js

# error: Uncaught ReferenceError: isBoolean is not defined
RUN deno run ./index.js
```

If I use `hayd/deno:alpine-1.4.6` as base image, everything works file

**Comment by Nitive** [comment]
Same problem with deno@1.5.3

```sh
$ deno --version
deno 1.5.3 (ab7b8ba, release, x86_64-apple-darwin)
v8 8.7.220.3
typescript 4.0.5

$ deno bundle index.ts index.js
Bundle file:///Users/nitive/Work/csssr-infrastructure/deno/tools/test/index.ts
Check file:///Users/nitive/Work/csssr-infrastructure/deno/tools/test/index.ts
Emit "index.js" (91.53KB)

$ deno run index.js
error: Uncaught ReferenceError: isBoolean is not defined
    predicate: isBoolean,
               ^
    at file:///Users/nitive/Work/csssr-infrastructure/deno/tools/test/index.js:136:16
```

**Comment by MarkTiedemann** [comment]
I'm hitting a similar error when bundling a specific import rather than `*`.

```typescript
import { stringify } from "https://deno.land/std@0.78.0/encoding/yaml.ts";
console.log(stringify({ a: 1 }));
```

```
> deno run index.js
error: Uncaught ReferenceError: Cannot access 'failsafe' before initialization
        failsafe
        ^
    at file:///C:/dev/test/index.js:2289:9
```

**Comment by kitsonk** [comment]
cc/ @kdy1

**Comment by bartlomieju** [comment]
Fixed in #8443

**Comment by Nitive** [comment]
Thank you! Everything is working on v1.5.4
